### PR TITLE
require to use extensions in import specifiers

### PR DESCRIPTION
**Pull request by aduh95**
This fixes an error where bundlers would report a `ModuleNotFound` error because the extensions are missing:

https://codesandbox.io/s/nifty-fire-7luyeb?file=/package.json:252-350

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,9 +4,10 @@
   ],
   "env": {
     "es6": true,
-    "node": true,
+    "node": false,
+    "jest": false,
     "browser": true,
```
Comment (Acconut) [inline .eslintrc:9]: Can we also enable the browser env only for files in `lib/browser/`?

File: demos/nodejs/index.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 /* eslint-disable no-console */
 
+'use strict'
+
 const fs = require('fs')
-const tus = require('../..')
+const tus = require('../..') // eslint-disable-line import/no-unresolved
```
Comment (Acconut) [inline demos/nodejs/index.js:6]: What is this eslint comment about? The files should be present once the build scripts has been run?

File: demos/browser/demo.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 /* global tus */
 /* eslint-disable no-console, no-alert */
 
+'use strict'
```
Comment (Acconut) [inline demos/browser/demo.js:4]: Why do some files get strict mode but not all?

**Review by Acconut — CHANGES_REQUESTED** [review]

File: demos/nodejs/index.js (hunk 2 above)
Comment (aduh95) [reply demos/nodejs/index.js:6]: Ah that might be because I didn't build before running the linter indeed.

File: demos/browser/demo.js (hunk 3 above)
Comment (aduh95) [reply demos/browser/demo.js:4]: See https://tc39.es/ecma262/#sec-strict-mode-code: 

> [Module code](https://tc39.es/ecma262/#sec-types-of-source-code) is always strict mode code.

That means, only non-ESM files need the `'use strict'` directive.

File: .eslintrc (hunk 1 above)
Comment (aduh95) [reply .eslintrc]: Good call. That's done, the linter does complain about the use of `window`, but nothing a `// eslint-disable-line` comment can't fix.

File: demos/browser/demo.js (hunk 3 above)
Comment (Acconut) [reply demos/browser/demo.js:4]: I see, that you!

**Review by Acconut — APPROVED** [review]
Thank you!

**Comment by Acconut** [comment]
I just released 3.0.1 with these patches.

**Pull request merged**